File: Bin/Data/Templates/CnIniFiler.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "<#UnitName>.h"
//---------------------------------------------------------------------------
T<#IniClassName> *<#IniClassName>;

<#IniSections>
<#IniNames>//---------------------------------------------------------------------------

void T<#IniClassName>::LoadSettings(TMemIniFile *Ini)
{
  if (Ini != NULL)
  {
<#IniReaders>  }
}

void T<#IniClassName>::SaveSettings(TMemIniFile *Ini)
{
  if (Ini != NULL)
  {
<#IniWriters>  }
}

void T<#IniClassName>::LoadFromFile(const AnsiString FileName)
{
  TMemIniFile *Ini;
  Ini = new TMemIniFile(FileName);
    
  __try
  {
    LoadSettings(Ini);
  }
  __finally
  {
    delete Ini;
  }
}

void T<#IniClassName>::SaveToFile(const AnsiString FileName)
{
  TMemIniFile *Ini;
  Ini = new TMemIniFile(FileName);
  
  __try
  {
    SaveSettings(Ini);
    Ini->UpdateFile();
  }
  __finally
  {
    delete Ini;
  }
}

#pragma package(smart_init)

File: Bin/Data/Templates/CnIniFiler.pas
unit <#UnitName>;

interface

uses
  Classes, SysUtils, IniFiles, Forms, Windows;

const
<#IniSections>
<#IniNames>type
  T<#IniClassName> = class(TObject)
  private
<#IniFields>  public
    procedure LoadSettings(Ini: TMemIniFile);
    procedure SaveSettings(Ini: TMemIniFile);
    
    procedure LoadFromFile(const FileName: string);
    procedure SaveToFile(const FileName: string);

<#IniProperties>  end;

var
  <#IniClassName>: T<#IniClassName> = nil;

implementation

procedure T<#IniClassName>.LoadSettings(Ini: TMemIniFile);
begin
  if Ini <> nil then
  begin
<#IniReaders>  end;
end;

procedure T<#IniClassName>.SaveSettings(Ini: TMemIniFile);
begin
  if Ini <> nil then
  begin
<#IniWriters>  end;
end;

procedure T<#IniClassName>.LoadFromFile(const FileName: string);
var
  Ini: TMemIniFile;
begin
  Ini := TMemIniFile.Create(FileName);
  try
    LoadSettings(Ini);
  finally
    Ini.Free;
  end;
end;

procedure T<#IniClassName>.SaveToFile(const FileName: string);
var
  Ini: TMemIniFile;
begin
  Ini := TMemIniFile.Create(FileName);
  try
    SaveSettings(Ini);
    Ini.UpdateFile;
  finally
    Ini.Free;
  end;
end;

initialization
  <#IniClassName> := T<#IniClassName>.Create;

finalization
  <#IniClassName>.Free;

end.

